streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala

```diff
@@ -0,0 +1,570 @@
+    inner.branch(predicates.map(_.asPredicate): _*).map(kstream => wrapKStream(kstream))
+  }
+
+  /**
+   * Materialize this stream to a topic and creates a new [[KStream]] from the topic using the `Produced` instance for 
+   * configuration of the `Serde key serde`, `Serde value serde`, and `StreamPartitioner`
+   * <p>
+   * The user can either supply the `Produced` instance as an implicit in scope or she can also provide implicit
+   * key and value serdes that will be converted to a `Produced` instance implicitly.
+   * <p>
+   * {{{
+   * Example:
+   *
+   * // brings implicit serdes in scope
+   * import DefaultSerdes._
+   *
+   * //..
+   * val clicksPerRegion: KTable[String, Long] = //..
+   *
+   * // Implicit serdes in scope will generate an implicit Produced instance, which
+   * // will be passed automatically to the call of through below
+   * clicksPerRegion.through(topic)
+   *
+   * // Similarly you can create an implicit Produced and it will be passed implicitly
+   * // to the through call
+   * }}}
+   *
+   * @param topic the topic name
+   * @param (implicit) produced the instance of Produced that gives the serdes and `StreamPartitioner`
+   * @return a [[KStream]] that contains the exact same (and potentially repartitioned) records as this [[KStream]]
+   * @see `org.apache.kafka.streams.kstream.KStream#through`
+   */
+  def through(topic: String)(implicit produced: Produced[K, V]): KStream[K, V] =
+    inner.through(topic, produced)
+
+  /**
+   * Materialize this stream to a topic using the `Produced` instance for 
+   * configuration of the `Serde key serde`, `Serde value serde`, and `StreamPartitioner`
+   * <p>
+   * The user can either supply the `Produced` instance as an implicit in scope or she can also provide implicit
+   * key and value serdes that will be converted to a `Produced` instance implicitly.
+   * <p>
+   * {{{
+   * Example:
+   *
+   * // brings implicit serdes in scope
+   * import DefaultSerdes._
+   *
+   * //..
+   * val clicksPerRegion: KTable[String, Long] = //..
+   *
+   * // Implicit serdes in scope will generate an implicit Produced instance, which
+   * // will be passed automatically to the call of through below
+   * clicksPerRegion.to(topic)
+   *
+   * // Similarly you can create an implicit Produced and it will be passed implicitly
+   * // to the through call
+   * }}}
+   *
+   * @param topic the topic name
+   * @param (implicit) produced the instance of Produced that gives the serdes and `StreamPartitioner`
+   * @see `org.apache.kafka.streams.kstream.KStream#to`
+   */
+  def to(topic: String)(implicit produced: Produced[K, V]): Unit =
+    inner.to(topic, produced)
+
+  /**
+   * Transform each record of the input stream into zero or more records in the output stream (both key and value type
+   * can be altered arbitrarily).
+   * A `Transformer` (provided by the given `TransformerSupplier`) is applied to each input record and
+   * computes zero or more output records.
```

Comment:
Maybe add a sentence, that stores must be added via `addStateStore` or `addGlobalStore` before they can be connected to the `Transformer` ?

Reply:
done ..